[5.0] Improve ext-mbstring Requirements In composer.json Files

vlakoff opened the pull request:
Thankfully, a simple search for "mb_" in files does the job.

By the way, Stringy compatibility has been improved (refs https://github.com/danielstjules/Stringy/pull/57).

GrahamCampbell commented on src/Illuminate/Filesystem/composer.json:
```diff
@@ -15,6 +15,9 @@
         "symfony/finder": "2.6.*",
         "league/flysystem": "~1.0"
     },
+    "require-dev": {
```
please remove this

GrahamCampbell commented:
I'm a little confused by these changes. Also, you haven't updated the support package.

vlakoff commented:
Already present in support package.

For the rest, I think it's mainly about modularity vs (maybe) maintainability.
I gave you the elements, now I let you decide :)

vlakoff commented:
The require-*dev* for mbstring in Filesystem package could easily be removed. Only [one line](https://github.com/laravel/framework/blob/44623342bc3907eab43860eb2f0cacdc59ef09fa/tests/Filesystem/FilesystemTest.php#L174) (recently introduced in #6905) needs mbstring, and is easy to make work without mbstring. It's just that it would be conflicting with this PR.

GrahamCampbell commented on composer.json:
```diff
@@ -13,9 +13,8 @@
         "php": ">=5.4.0",
         "ext-openssl": "*",
         "ext-mcrypt": "*",
-        "ext-mbstring": "*",
```
keep this if you're not removing it in the support package

vlakoff replied:
I don't get it... Support package definitely needs the ext, why would I remove it? Then, isn't the ext redundant in the main package?

GrahamCampbell replied:
No. The main composer.json doesn't inherit the others. It just totally replaces them. They aren't even looked at by composer.

vlakoff replied:
Ahh, right. Fixing.

vlakoff committed:
Update Stringy requirement.

Previous versions could bring some Composer-related
troubles if ext-mbstring was missing.

vlakoff committed:
Require ext-mbstring in packages that use it.

GrahamCampbell commented on src/Illuminate/Auth/composer.json:
```diff
@@ -10,6 +10,7 @@
     ],
     "require": {
         "php": ">=5.4.0",
+        "ext-mbstring": "*",
```
please remove this

GrahamCampbell commented on src/Illuminate/Filesystem/composer.json:
```diff
@@ -15,6 +15,9 @@
         "league/flysystem": "~1.0"
     },
+    "require-dev": {
+        "ext-mbstring": "*",
```
please remove this

GrahamCampbell commented:
This makes no sense.

GrahamCampbell commented:
If support needs the extension then so do the others. They inherit the dependency. It doesn't need adding again.

vlakoff commented:
I was a bit in a hurry and missed that obvious point. My apologies. So you can simply update both stringy requirements (just to be on the safe side) and everything's fine.
